<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:BW="java://com.tibco.pe.core.JavaCustomXPATHFunctions" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns2="http://three.co.uk/xsd/resource/common/commondefinitions/msf/messagecontext/v1" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns1="http://www.tibco.com/xsd/orderfallout/businessexceptionmanagement/orderfalloutbusinessexception.xsd" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:ns4="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:ns3="http://three.co.uk/xsd/resource/common/commondefinitions/msf/exception/v1" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:pfx4="http://www.tibco.com/CommonLE2/namespace/Public/Exception/ResolutionReply.xsd" xmlns:pfx3="http://xmlns.example.com/BEHResolution_v2" xmlns:pfx2="http://xmlns.example.com/BEHResolution_v1" xmlns:ns5="http://three.co.uk/xsd/resource/common/commondefinitions/msf/exceptioninfo/v1" xmlns:ns6="http://www.tibco.com/schemas/MSF_OrderFallout/SharedResources/SchemaDefinitions/External/BussnessExceptionHandler/ResolutionSchema.xsd" xmlns:pfx="http://xmlns.example.com/1418045315253">
    <xs:import xmlns:xs="http://www.w3.org/2001/XMLSchema" namespace="http://www.tibco.com/CommonLE2/namespace/Public/Exception/ResolutionReply.xsd" schemaLocation="/SharedResources/Schemas/Public/Exception/ResolutionReply.xsd"/>
    <wsdl:import namespace="http://xmlns.example.com/BEHResolution_v1" location="/SharedResources/WebServiceDefinitions/External/BEH/BEHResolution_v1.wsdl"/>
    <pd:name>IntegrationServices/OrderFallout/BusinessExceptionResolution_v1/Service/PortType/wsPQFResolution.process</pd:name>
    <pd:startName>Start</pd:startName>
    <pd:startType wsMsgRef="pfx2:PQFResolutionMessage"/>
    <pd:startX>30</pd:startX>
    <pd:startY>63</pd:startY>
    <pd:returnBindings>
        <pfx3:Output/>
    </pd:returnBindings>
    <pd:endName>End</pd:endName>
    <pd:endX>869</pd:endX>
    <pd:endY>63</pd:endY>
    <pd:errorSchemas/>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1355136884829</pd:targetNamespace>
    <pd:activity name="Audit End">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>620</pd:x>
        <pd:y>63</pd:y>
        <config>
            <processName>/UtilityProcesses/Audit/LogAuditEnd.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <response>
                    <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage/resolution"/>
                </response>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Audit Start">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>379</pd:x>
        <pd:y>63</pd:y>
        <config>
            <processName>/UtilityProcesses/Audit/LogAuditStart.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <payload>
                    <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage/ancestor-or-self::*/namespace::node()"/>
                    <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage/@*"/>
                    <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage/node()"/>
                </payload>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Catch All">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>109</pd:x>
        <pd:y>305</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <catchAll>true</catchAll>
            <fault>localname=Exception namespace=http://three.co.uk/xsd/resource/common/commondefinitions/msf/exception/v1</fault>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Catch Exception">
        <pd:type>com.tibco.pe.core.CatchActivity</pd:type>
        <pd:resourceType>ae.activities.catch</pd:resourceType>
        <pd:x>250</pd:x>
        <pd:y>445</pd:y>
        <pd:handler>true</pd:handler>
        <config>
            <fault>localname=Exception namespace=http://three.co.uk/xsd/resource/common/commondefinitions/msf/exception/v1</fault>
            <catchAll>false</catchAll>
        </config>
        <pd:inputBindings/>
    </pd:activity>
    <pd:activity name="Create Exception Catch All">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>250</pd:x>
        <pd:y>313</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/CreateException_CatchAll.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$_error/ns:ErrorReport"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Create Exception Receive Error">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>250</pd:x>
        <pd:y>185</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/CreateException_ReceiveError.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$_error/ns:ErrorReport"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Initialise">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>138</pd:x>
        <pd:y>63</pd:y>
        <config>
            <processName>/UtilityProcesses/StartUp/Initialise.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <InitialProcessName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </InitialProcessName>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Initialise Receive Error">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>110</pd:x>
        <pd:y>185</pd:y>
        <config>
            <processName>/UtilityProcesses/StartUp/Initialise.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <InitialProcessName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </InitialProcessName>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Log Exception Catch All">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>391</pd:x>
        <pd:y>313</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/LogException.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$Create-Exception-Catch-All/root/ns3:Exception"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Log Exception Catch Exception">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>390</pd:x>
        <pd:y>453</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/LogException.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$Catch-Exception/ns3:Exception"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Log Exception Receive Error">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>384</pd:x>
        <pd:y>185</pd:y>
        <config>
            <processName>/UtilityProcesses/Exceptions/LogException.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <xsl:copy-of select="$Create-Exception-Receive-Error/root/ns3:Exception"/>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="MainHandlePQFResoultionRequest">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>500</pd:x>
        <pd:y>63</pd:y>
        <config>
            <processName>/IntegrationServices/OrderFallout/BusinessExceptionResolution_v1/Main/MainHandlePQFResoultionRequest.process</processName>
        </config>
        <pd:inputBindings>
            <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage"/>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Trace End">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>752</pd:x>
        <pd:y>63</pd:y>
        <config>
            <processName>/UtilityProcesses/Tracing/TraceEnd.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <processName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </processName>
                <xsl:if test="$Start/pfx2:PQFResolutionMessage/resolution/ns1:PQF_BusinessExceptionResponse/ns1:resolution">
                    <body>
                        <xsl:value-of select="string($Start/pfx2:PQFResolutionMessage/resolution/ns1:PQF_BusinessExceptionResponse)"/>
                    </body>
                </xsl:if>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="Trace Start">
        <pd:type>com.tibco.pe.core.CallProcessActivity</pd:type>
        <pd:resourceType>ae.process.subprocess</pd:resourceType>
        <pd:x>252</pd:x>
        <pd:y>63</pd:y>
        <config>
            <processName>/UtilityProcesses/Tracing/TraceStart.process</processName>
        </config>
        <pd:inputBindings>
            <root>
                <processName>
                    <xsl:value-of select="BW:getCurrentProcessName($_processContext/ns:ProcessContext/ProcessId)"/>
                </processName>
                <payload>
                    <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage/ancestor-or-self::*/namespace::node()"/>
                    <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage/@*"/>
                    <xsl:copy-of select="$Start/pfx2:PQFResolutionMessage/node()"/>
                </payload>
            </root>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Initialise</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Initialise</pd:from>
        <pd:to>Trace Start</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Audit Start</pd:from>
        <pd:to>MainHandlePQFResoultionRequest</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Trace End</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Initialise Receive Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-65536</pd:lineColor>
        <pd:conditionType>error</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Initialise Receive Error</pd:from>
        <pd:to>Create Exception Receive Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Create Exception Receive Error</pd:from>
        <pd:to>Log Exception Receive Error</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch All</pd:from>
        <pd:to>Create Exception Catch All</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Create Exception Catch All</pd:from>
        <pd:to>Log Exception Catch All</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Catch Exception</pd:from>
        <pd:to>Log Exception Catch Exception</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Trace Start</pd:from>
        <pd:to>Audit Start</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>MainHandlePQFResoultionRequest</pd:from>
        <pd:to>Audit End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Log Exception Receive Error</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>One Bend</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Log Exception Catch All</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>One Bend</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Log Exception Catch Exception</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>One Bend</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Audit End</pd:from>
        <pd:to>Trace End</pd:to>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
</pd:ProcessDefinition>